enclave_wrangler/dataset_upload.py

```diff
@@ -34,11 +35,131 @@
+        cset_name = container_d['concept_set_id']
+        linked_csets[cset_name] = {'versions': {}}
+        linked_csets[cset_name]['container'] = container_d
+        code_sets_df_i = code_sets_df[code_sets_df['concept_set_name'] == cset_name]
+        code_sets_d_i = json.loads(code_sets_df_i.to_json(orient='records'))
+        for version_d in code_sets_d_i:
+            version_id = version_d['codeset_id']
+            linked_csets[cset_name]['versions'][version_id] = {'items': []}
+            linked_csets[cset_name]['versions'][version_id]['version'] = version_d
+            concept_set_version_item_rv_edited_df_i = concept_set_version_item_rv_edited_df[
+                concept_set_version_item_rv_edited_df['codeset_id'] == version_id]
+            concept_set_version_item_rv_edited_d_i = json.loads(concept_set_version_item_rv_edited_df_i.to_json(orient='records'))
+            for item_d in concept_set_version_item_rv_edited_d_i:
+                linked_csets[cset_name]['versions'][version_id]['items'].append(item_d)
+
+    # Upload to enclave
+    for linked_cset in linked_csets.values():
+        container_d = linked_cset['container']
+        response = upload_concept_set(  # concept_set_container
+            concept_set_id=container_d['concept_set_name'],
+            intention=container_d['intention'],
+            # todo: research_project: (a) default this to ENCLAVE_PROJECT_NAME in func, (b) do that here, (c) add it as a column to an
+            #  updated palantir-3-file for the new api
+            research_project=ENCLAVE_PROJECT_NAME,
+            assigned_sme=PALANTIR_ENCLAVE_USER_ID_1,
+            assigned_informatician=PALANTIR_ENCLAVE_USER_ID_1,
+            # TODO: validate first (if this func says validate_first=True, then upload
+            validate=True)
+
+        for version in linked_cset['versions'].values():
+            version_d = version['version']
+            response = upload_draft_concept_set(  # code_set
+                # todo: domain_team: @Siggie: Not sure what to put here, but it is optional param, so I'm leaving blank.
+                # domain_team=version_d[''],
+                provenance=version_d['provenance'],
+                # todo: current-max-version: @Siggie: Sucks that we have to set this. but probaby because version types
+                #  can be arbitrary, so enclave can't know. but this means we have to have a local registry for most
+                #  recent version as well for anything we're uploading, or get it when we re-download, right?
+                #  ...docs, for reference: This must be set to the current maximum version number assigned to a version
+                #  of this concept set, or null if creating the first version of a concept set. If null, then
+                #  baseVersion is not "required".
+                #  ...Leaving this blank for now since is optional param.
+                # current_max_version=version_d[''],
```

Comment:
Leaving `current-max-version` blank because I don't have a way to reliably ascertain. More comments about this in the code.